migrations/versions/1210_add_process_outcome_model.py

```diff
@@ -0,0 +1,126 @@
+"""Add Award model
+
+Revision ID: 1150
+Revises: 1140
+Create Date: 2018-05-10 17:22:50.773017
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+
+# revision identifiers, used by Alembic.
+revision = '1210'
+down_revision = '1200'
+
+
+def upgrade():
+    op.create_unique_constraint('uq_brief_responses_id_brief_id', 'brief_responses', ['id', 'brief_id'])
+    op.create_unique_constraint(
+        'uq_direct_award_search_result_entries_archived_service_id_search_id',
+        'direct_award_search_result_entries',
+        ['archived_service_id', 'search_id'],
+    )
+    op.create_unique_constraint('uq_direct_award_searches_id_project_id', 'direct_award_searches', ['id', 'project_id'])
+    op.create_table('process_outcomes',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('external_id', sa.BigInteger(), nullable=False),
+        sa.Column('completed_at', sa.DateTime(), nullable=True),
+        sa.Column('start_date', sa.Date(), nullable=True),
+        sa.Column('end_date', sa.Date(), nullable=True),
+        sa.Column('awarding_organisation_name', sa.String(), nullable=True),
+        sa.Column('award_value', sa.Numeric(precision=11, scale=2), nullable=True),
+        sa.Column('result', sa.String(), nullable=False),
+        sa.Column('direct_award_project_id', sa.Integer(), nullable=True),
+        sa.Column('direct_award_search_id', sa.Integer(), nullable=True),
+        sa.Column('direct_award_archived_service_id', sa.Integer(), nullable=True),
+        sa.Column('brief_id', sa.Integer(), nullable=True),
```

Comment:
I don't think these should really be in there yet - they aren't used, and having them existing would be confusing, assuming we aren't going to start work on this straight away.

Which I think is a given - we would only refactor the existing briefs stuff when we next do major work on that flow, which isn't even slightly in a roadmap right now. They can stay, but commented out - but I realise that has implications for the tests you're writing. We shouldn't cover the briefs case at all for now, I think.

Happy to chat through on slack/hangouts/whatever if you like.

Reply:
We came up with a solution that works in a bulletproof way for both DAs and Briefs (through quite a bit of grief and debate), we should implement it. Not implement the braindead easy half and then leave it for the next person (perhaps not you or I) to come along and not quite get what the plan was and implement it in an asymmetric not-quite-joined-up way. I also think this is (certainly) one of those cases where making it "work" (to a releasable state) in the easy way and *then* the hard way is *significantly* more work than just making it all work straight off. Which it does. Look, it's here.

If we were just going to implement one half of this and then leave it with a question mark over it we could have completely skipped our wonderful debate.

Reply:
The point of the debate was so that we agreed what we want this thing to look like - as a roadmap. So that there's a plan to follow, when we next touch the DOS code, to make it not be gloriously inconsistent. But yes I regret having that debate at all considering the confusion it has clearly caused - I honestly thought we were on the same page, sorry :-(

And it doesn't mean I want to have a load of code in the codebase that never gets run, and sits around as a weird stub - that would be _more_ confusing for people.

I take your point about people not-quite-getting the plan... but if the plan is good and clear, then really they should! Which is why the write-up phase of this is important (and I committed to getting that done). And I'd rather have it commented out than lose the work, there's definitely value there.

Reply:
In short, I went to the trouble of designing a solution for this and *making sure that it does actually work*, traversing any kinks involved, so I *must* make some kind of concrete manifestation of it, otherwise it was a waste of time.

Reply:
Understood, that's fine. Let's get this committed - most of my other comments still stand although are fairly minor, so once you've been through them, and someone else has approved the PR too, then I'm all like 👍

Reply:
I'm with @galund on this one. If we're not using it yet we shouldn't include it.

Reply:
Re this comment:
```
    # NOTE though ProcessOutcome currently has the *ability* to be associated with a Brief/BriefResponse, at time of
    # writing this is not yet *used* for reporting brief awards - those are still using their own mechanism, but should
    # be ported over to this mechanism as soon as time allows.
```
I think this needs more discussion re a backlog/ tech debt ticket. We shouldn't dictate future decisions in the codebase like this. It may be de prioritised or even completely removed in the future and then the comment is untrue. 
The platform will still operate if this bit of work doesn't get done.

Reply:
Well, I think it's okay to dictate design/architecture - that means we've done some thinking.

In this case, we needed to implement half of that piece of architecture, and some implementation decisions then have to be taken - which necessarily constrains that future work.

We could leave the thing out - but that doesn't actually take away from the fact that we've _already_ constrained the future implementation (because otherwise the existing implementation of the architecture doesn't make sense).

I don't like having code that doesn't do anything in there either. We've definitely taken on some tech debt here - although it's not super-bad - as you say, everything carries on working, it just doesn't have the greatest architecture. But that was true _before we did any work for G-Cloud_: the only alternative was a mechanism that collected outcomes for G-Cloud that didn't account for further competition at all _anyway_. There was never going to be a scenario where we could actually move over DOS and implement G-Cloud in one fell swoop - it's too big a piece of work.

Some I'm content that there's really no great choices here, and the working having been done means we can still choose to:

(a) do nothing for now - the situation is commented
(b) move DOS over as planned here and more-than-partially implemented
(c) decide some or other implementation decision here is wrong and come up with an entirely new idea for capturing this stuff.

Personally, long run, I don't think this is even the right platform for capturing outcomes - that's what Contracts Finder is for - but we are constrained by the organisational problems we all know about.